### PR TITLE
Profile Controllers

### PR DESCRIPTION
**Pull request by rishabdhamija**
### What this PR does (required):
- created server/routes/profile.js
- created server/controllres/profile.js with the business logic associated for the routes in profile.js file.
- created test/profileDB.js file with dummy data used for testing (consists of 4 profiles)

### Screenshots / Videos (required):
Below are screenshots for 4 routes tested using postman:
- POST("/profile")
Screen Shot 2021-06-14 at 6 36 09 PM

- GET ("/profile/:id")
Screen Shot 2021-06-14 at 6 55 25 PM

- GET ("/profile")
Screen Shot 2021-06-14 at 6 53 54 PM

- UPDATE ("/profile/:id")
Screen Shot 2021-06-14 at 6 54 55 PM

### Any information needed to test this feature (required):
- the above screenshots test all of the 4 cases associated with profileControllers.js file
- 
### Any issues with the current functionality (optional):
-  The MongoDB URI in the .env file represents the  url "mongodb+srv://team-shiba:team-shib%40@cluster0.s4x0p.mongodb.net/teamShibaDB?retryWrites=true&w=majority"

### CODE REVIEW

File: server/controllers/profile.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+const profileModel = require('../models/Profile');
+
+exports.postProfile = asyncHandler(async (req, res, next) => {
```
Comment (bonnieli) [inline server/controllers/profile.js:5]: what does `postProfile` mean? does it mean to create a new profile? if so i would rename this to `createProfile`

File: server/controllers/profile.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  const { firstname, lastname, description, availability } = req.body;
```
Comment (bonnieli) [inline server/controllers/profile.js:6]: maybe add some validation that these are required

**Review by bonnieli — COMMENTED** [review]
Great separation of PRs

File: server/controllers/profile.js (hunk 1 above)
Comment (rishabdhamija) [reply server/controllers/profile.js:5]: changed the name from postProfile to createProfile.

File: server/controllers/profile.js (hunk 2 above)
Comment (rishabdhamija) [reply server/controllers/profile.js:6]: I already have input validation established in models/Profile.js for input length as  well as availability being from a certain set of values. I didn't want to repeat validation inside the controller also. Please let me know if required then I will remove validation from mongoose schema and put it in the controller.

**Review by bonnieli — APPROVED** [review]

**Pull request merged**